doc/doc-primitives.saty

```diff
@@ -310,17 +427,47 @@ document (|
       +command (`set-text-color`) (tCLR --> (tCTX --> tCTX)) {
         \code{${ctx} \|\> set-text-color ${color}}で文字色を\code{${color}}に変更した文脈を返す．
       }
+      +command (`get-text-color`) (tCTX --> tCLR) {
+        テキスト文脈が保持している文字色を返す．
+      }
       +command (`set-leading`) (tL --> (tCTX --> tCTX)) {
         \code{${ctx} \|\> set-leading ${l}}で行送りを\code{${l}}に変更した文脈を返す．
         これはフォントサイズに対する比での指定ではなく，直接長さを指定するプリミティヴであることに注意．
         すなわち，フォントサイズを変更しても標準の行送りの長さは変更されない．
       }
+      +command (`set-min-gap-of-lines`) (tL --> (tCTX --> tCTX)) {
+        \code{${ctx} \|\> set-min-gap-of-lines ${l}}で行間の最小値を引数の値に変更した文脈を返す．
+      }
+      +command (`set-paragraph-margin`) (tL --> tL --> ((tCTX --> tCTX))) {
+        \code{${ctx} \|\> set-min-gap-of-lines ${l_1} ${l_2}}で段落の上のマージンを\code{${l_1}}に，
+        段落の下のマージンを\code{${l_2}}にするように変更したテキスト文脈を返す．
+        \subject-to-change;
+      }
       +command (`set-manual-rising`) (tL --> (tCTX --> tCTX)) {
         \code{${ctx} \|\> set-manual-rising ${l}}で文字全体を長さ\code{${l}}だけ持ち上げて組む文脈を返す．
       }
       +command (`get-text-width`) (tCTX --> tL) {
         テキスト文脈が保持している段落幅を返す．\code{line-break}はこの長さにしたがって行分割を行なう．
       }
+      +command (`set-word-break-penalty`) (tI --> (tCTX --> tCTX)) {
+        \code{set-word-break-penalty ${p} ${ctx}}で“そこで行分割された場合に加算されるペナルティ値”を\code{${p}}に設定したテキスト文脈を返す．
+      }
+      +command (`set-every-word-break`) (tIB --> (tIB --> (tCTX --> tCTX))) {
+        \code{set-every-word-break ${ib_1} ${ib_2} ${ctx}}で行分割候補箇所の“そこで行分割された場合に直前の行末に入る内容”を\code{${ib_1}}に，
+        “そこで行分割された場合に直後の行頭に入る内容”を\code{${ib_2}}に設定したテキスト文脈を返す．
+      }
+      +command (`get-every-word-break`) (tCTX --> (tPROD [tIB; tIB])) {
+        \code{set-every-word-break}で設定されている\code{${ib_1}}と\code{${ib_2}}の内容を取得する．
+      }
+      +command (`set-hyphen-penalty`) (tI --> (tCTX --> tCTX)) {
+        ハイフネーションによるペナルティを設定する．
+        デフォルトは100である．
+      }
+      +command (`set-hyphen-min`) (tI --> (tI --> (tCTX --> tCTX))) {
+        \code{${ctx} \|\> set-hyphen-min ${p_1} ${p_2}}で
+        左のハイフネーションペナルティを\code{${p_1}}に，
+        右のハイフネーションペナルティを\code{${p_2}}に設定したテキスト文脈を返す．
+      }
```

Comment:
This explanation looks somewhat different from the actual behavior; IMHO it should be like:

```
\code{${ctx} \|\> set-hyphen-min ${n_1} ${n_2}} で
ハイフネーションによって単語が分割されるときの左側の最小文字数を \code{${n_1}} に，右側の最小文字数を \code{${n_2}} にそれぞれ設定したテキスト文脈を返す．
```